lib/Lib.thy

```diff
@@ -2761,6 +2761,19 @@ lemma distinct_map_enum:
    \<Longrightarrow> distinct (map F (enum_class.enum :: 'a :: enum list))"
   by (simp add: distinct_map inj_onI)
 
+\<comment>\<open>
+  Helps transform facts with multiple instances of a variable so that
+  they match goals up to equality.
+
+  Why does this lemma exist when lemmas like @{thm subst} already exist?
+  Because patterns like `?P ?x` usually have multiple unifiers with the
+  conclusion of a rule, whereas `(\<And>x. ?P x x)` only has one. This means
+  you can "inline" `equify` like this: `my_cool_thm[THEN equify]`.
+\<close>
+lemma equify:
+  "(\<And>x. P x x) \<Longrightarrow> x = x' \<Longrightarrow> P x x'"
+  by simp
```

Comment:
is this an MCS-only thing? I'm not seeing it on master. It looks like it has Ed-style comment indentation (including the extra `>` on its own line which we don't do anymore)

Reply:
Yeah, it was added on MCS and then I accidentally removed it during the merge.

I can change the indentation as part of the merge if you'd prefer. Would it be something like this?
```suggestion
―‹Helps transform facts with multiple instances of a variable so that
  they match goals up to equality.

  Why does this lemma exist when lemmas like @{thm subst} already exist?
  Because patterns like `?P ?x` usually have multiple unifiers with the
  conclusion of a rule, whereas `(⋀x. ?P x x)` only has one. This means
  you can "inline" `equify` like this: `my_cool_thm[THEN equify]`.›
```

Reply:
Probably for a `lemma` we should stick to `text` or `(* *)` ... so for this one, I'd be fine with only moving the `>` to the previous line so it isn't on its own.